### DISCUSSION TITLE
Why is there still no full support for C/C++ projects?

### DISCUSSION THREAD

**Discussion by YingYan** [apache/netbeans#8015] [General]
It's all in the title.
I use NetBeans for my C dev and I still can't find a proper support like NetBeans 8 could offer.
Do you have a date?

**Comment by JoachimRohde** [comment]
See https://github.com/apache/netbeans/issues/6979

**Comment by houranled** [comment]
Have you  found a proper support like NetBeans 8 could offer yet?

**Reply by YingYan** [reply]
I continue to use the plugin for NetBeans 8, but I find it has more and more bugs. For example, when I upgraded to the latest version of NetBeans, the compilation time for my projects has increased significantly. Using the debugger is becoming complicated...

When the Apache Foundation took over, there was talk of rewriting this plugin, but there's no further news :(

**Reply by matthiasblaesing** [reply]
> When the Apache Foundation took over, there was talk of rewriting this plugin, but there's no further news :(

"The Apache Foundation" is a group of people investing their time to fix their and improve the IDE for their fields of work. That might or might not match your problem set. The only way to fix the "it does not work" problem is by investing your self.

**Reply by YingYan** [reply]
I would have liked to, but unfortunately I don't have the necessary skills to help.

But it was just a question, and I understand perfectly well that C/C++ development isn't a priority.

I really don't want to use another IDE, and with my advanced age, it's difficult to change one's habits :)

So I'll wait and keep hoping.

**Reply by muvicado** [reply]
I have a similar use case, and would like to see C/C++ support migrate forward.
I'm still using 8.2 in paranoia of losing things I love about that version.
Is there a list of differences between that large span of releases? I suppose I could just cat and grep the release notes, but is there an existing compilation?

**Reply by houranled** [reply]
@muvicado  NetBeans 8.2 is just too good to use. I have never found an IDE that can surpass or replace it.

**Reply by JoachimRohde** [reply]
> Is there a list of differences between that large span of releases? I suppose I could just cat and grep the release notes, but is there an existing compilation?

https://github.com/apache/netbeans/releases list changes back to version 9.0

**Reply by emaphis** [reply]
It's getting too long to remember now, but was the C++ module ever donated?

**Reply by matthiasblaesing** [reply]
Parts were donated. There is a working branch: https://github.com/apache/netbeans/tree/cnd
And some notes: https://cwiki.apache.org/confluence/display/NETBEANS/cnd+integration+proposal%3A+cnd-small

If I remember correctly the biggest showstopper is, that CND is based on CLANK a java translation of clang. That never matured and is abandondend.
